studio/src/window.cpp

```diff
@@ -46,10 +46,40 @@ namespace Studio {
 
 Window::Window(Arguments args)
     : QMainWindow()
-    , editor(new Editor(args.language))
     , view(new View)
     , settings("impraxical", "Studio")
 {
+    Language::Type language = Language::LANGUAGE_NONE;
+
+    if (args.language != Language::LANGUAGE_NONE)
+    {
+        // give precedence to command line arguments
+        language = args.language;
+    }
+    else if (settings.contains("language"))
+    {
+        // then application settings
+        QString language_str = settings.value("language").toString();
+        if (language_str == "guile")
+            language = Language::LANGUAGE_GUILE;
+        else if (language_str == "python")
+            language = Language::LANGUAGE_PYTHON;
+    }
+    // otherwise the editor will pick the default
+
+    editor = new Editor(language);
+
+    auto storeLanguage = [this] {
```

Comment:
Does this _need_ to be a lambda function?  If not, I'd prefer a normal slot in the `Window` class, to minimize cleverness.

Reply:
It does not need to, just personal preference. My reasoning is that this way, all code related to reading/writing language to settings is localized to these 30 lines consecutive lines. 
But I understand your point of view as well, so I've changed it